substate-store-impls/src/hash_tree/tier_framework.rs

```diff
@@ -20,7 +20,7 @@ pub trait StoredNode {
     fn from_jmt_node(node: &Node<Self::Payload>, key: &TreeNodeKey) -> Self;
 }
 
-pub trait Tier {
+pub trait TierView {
```

Comment:
I was thinking it's sort of like a particular model on top of the database - really it's just a viewpoint of the tree formed by the root node at the start of a Tier.

But maybe view is overloaded in DB land. Happy with alternatives, e.g. Tier.

Reply:
The only surprise I had with "View" was that it updates the DB and itself (which is more than "viewing").
Tier sounds cool and short, but if I saw the "Tier" name alone, I'd expect a minimal enum (i.e. `Tier { Node, Partition, Substate }`), not a logic-heavy trait.

In Java we would have no doubts: `TierManager` :D

Here... maybe StateTreeTier? Just something that avoids future name-clashes if we ever decide to split other structures into tiers _(because it's such a good idea, right?)_.

Reply:
Yes, sounds good! Let's go with `StateTreeTier`!

Speaking of - perhaps we should rename the whole `hash_tree` folder to `state_tree`?

Reply:
> state_tree?

At this point I'd only be worried about ppl thinking that we store state in this tree.
The current situation is ofc imperfect too (we sometimes call it "hash tree", sometimes "state hash tree", and sometimes "state tree"). I hope one day we'll be forced to store _something_ more than hashes in that tree, and the most appropriate name will be more obvious.

(or we can invent something that gives it a unique identifier, while only vaguely capturing the [moving-target] use-cases... e.g. "history tree"?)

Reply:
Sure, that's a fair worry.

That said, we already have the transaction tree and receipt tree in the node which both store hashes, so in the same vein, calling it a state tree is reasonable I think :)